api/platform/platform_controller.go

```diff
@@ -58,21 +54,21 @@ func checkPlatformMandatoryProperties(platform *types.Platform) error {
 }
 
 // addPlatform handler for POST /v1/platforms
-func (ctrl *Controller) addPlatform(rw http.ResponseWriter, req *http.Request) error {
+func (ctrl *Controller) addPlatform(request *filter.Request) (*filter.Response, error) {
 	logrus.Debug("Creating new platform")
 
-	platform, errDecode := getPlatformFromRequest(req)
+	platform, errDecode := getPlatformFromRequest(request)
 	if errDecode != nil {
-		return errDecode
+		return nil, errDecode
 	}
 	if errMandatoryProperties := checkPlatformMandatoryProperties(platform); errMandatoryProperties != nil {
-		return types.NewErrorResponse(errMandatoryProperties, http.StatusBadRequest, "BadRequest")
+		return nil, filter.NewErrorResponse(errMandatoryProperties, http.StatusBadRequest, "BadRequest")
```

Comment:
When we return an error it will be converted to `Response` from the `HandleError` method in errors.go
I don't think we should construct a response object here.
Do you have a specific concern about this?